File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);
        int l = deck.length;
        int[] res = new int[l];
        ArrayList<Integer> ind = new ArrayList<>();
        
        for (int x=0; x<l; x++){
            ind.add(x);
        }
        
        for (int i=0; i<l; i++){
            if (i == l-1) break;
            else {
                int k = ind.remove(i+1);
                ind.add(k);
            }
        }
        
        for (int j=0; j<ind.size(); j++){
            int val = ind.get(j);
            res[val] = deck[j];
        }
        
        return res;
    }
}

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        num = sorted(deck)
        l = len(num)
        q = [0] * l
        ind = [i for i in range(l)]
        
        for i in range(l):
            if i == l-1:
                continue
            else:
                k = ind.pop(i+1)
                ind.append(k)
                
        for a,b in zip(ind, num):
            q[a] = b
            
        return q